backend/src/utils/helpers.py

```
@@ -0,0 +1,60 @@
+    digest = hashlib.sha256(ident.encode()).hexdigest()[:16]
+    lock_path = f"/tmp/periodic-{digest}.lock"
+
+    # Execution loop (use asyncio)
+    while True:
+        with process_lock(lock_path) as locked:
+            if locked:
+                try:
+                    job_func()
+
+                except Exception:
+                    raise
```

Ruff (0.14.14)

[warning] Remove exception handler; error is immediately re-raised

(TRY203)